### PR TITLE
HW2

### PR DESCRIPTION
**Pull request by andyalam**
- Wired up HW2 endpoints
- Added tests

### CODE REVIEW

File: hw2/src/main/java/com/example/hw2/service/CityService.java [hunk 1]
```diff
@@ -24,10 +27,17 @@
...
-		return null; 
+		List<City> cities = cityRepository.findByName(cityName);
+		if (cities.size() < 1) {
+			return null;
+		}
+
+		City city = cities.get(0);
+		Country country = countryRepository.findByCode(city.getCountryCode());
+		TempAndTime tempAndTime = weatherService.getTempAndTime(cityName);
+		String time = Long.toString(tempAndTime.time);
```
Comment (ROGERTERRILL) [inline hw2/src/main/java/com/example/hw2/service/CityService.java]: I would probably not have the time variable accessible publicly and instead use a method to access it and make time private. It was actually an issue I didn't see I had until I read your code.
```suggestion
		String time = Long.toString(tempAndTime.getTime());

```

File: hw2/src/main/java/com/example/hw2/service/CityService.java [hunk 2]
```diff
@@ -24,10 +27,17 @@
...
+
+		return new CityInfo(city, country.getName(), tempAndTime.temp, time);
```
Comment (ROGERTERRILL) [inline hw2/src/main/java/com/example/hw2/service/CityService.java]: Add a method to get the temp
```suggestion
		return new CityInfo(city, country.getName(), tempAndTime.getTemp(), time);

```

**Review by ROGERTERRILL — APPROVED** [review]
I only had two comments about how you are accessing some variables. I made the same choices but I think making your variables private and using public methods will protect your variables from being changed and you can return them in any format you want with a getter. Great job.

File: hw2/src/main/java/com/example/hw2/service/CityService.java (hunk 1 above)
Comment (andyalam) [reply hw2/src/main/java/com/example/hw2/service/CityService.java]: Implemented, thanks

File: hw2/src/main/java/com/example/hw2/service/CityService.java (hunk 2 above)
Comment (andyalam) [reply hw2/src/main/java/com/example/hw2/service/CityService.java]: Implemented, thanks

**Comment by ROGERTERRILL** [comment]
LGTG

**Comment by andyalam** [comment]
> LGTG

Looks good to go? Ok neat merging, thanks!

**Pull request merged**